### PR TITLE
Update session1/chapter7/tidb-ddl-status.md

### PR DESCRIPTION
**Pull request by tsthght**

### CODE REVIEW

File: session1/chapter7/tidb-ddl-status.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+### 1 TiDB DDL特点
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
### 1 TiDB DDL 特点
```

File: session1/chapter7/tidb-ddl-status.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+大多数数据库执行 DDL 操作时，或多或少会对正在访问该数据库的 SQL 产生影响。例如，在执行 DDL 期间可能会有锁表操作，此时访问该表的 SQL 会被阻塞。因此，一般在表结构设计阶段都会尽量避免后续产生 DDL 操作；如果必须执行 DDL操作，也只能选择在业务低峰期操作，尽量减少对线上业务的影响。
+
+TiDB 上的 DDL 操作，不会阻塞任何该数据库上正在执行的 SQL，对业务的 SQL 访问和对 DBA 运维都极为友好，这也是 TiDB 相较于其他数据库产品的一大优势所在。
+
+### 2 对 DDL 进行管理
+TiDB 对 MySQL 语法进行了扩展，通过 ADMIN 语句对 DDL 操作进行管理。下面罗列了 DDL 管理的基本命令，各命令返回结果中各字段的详细含义，请参考[官方文档](https://pingcap.com/docs-cn/stable/reference/sql/statements/admin/#admin)
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
TiDB 对 MySQL 语法进行了扩展，通过 ADMIN 语句对 DDL 操作进行管理。下面罗列了 DDL 管理的基本命令，各命令返回结果中各字段的详细含义，请参考[官方 admin 相关文档](https://pingcap.com/docs-cn/stable/reference/sql/statements/admin/#admin)
```

File: session1/chapter7/tidb-ddl-status.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+ADMIN SHOW DDL;
+# 查看当前 DDL 作业队列中所有结果(包括正在运行的 DDL 任务和等待运行的任务)
+ADMIN SHOW DDL JOBS;
+# 根据 JOB_ID 查询具体的 DDL 语句
+ADMIN SHOW DDL JOB QUERIES job_id [, job_id] ...;
+# 取消正在执行中的 DDL 任务
+ADMIN CANCEL DDL JOBS job_id [, job_id] ...;
+# 通过 JOB_ID 恢复表，等价于：RECOVER TABLE table_name
+RECOVER TABLE BY JOB ddl_job_id
+```
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md:50]: 将上面的语句分开会不会好一些。例如
查看当前 schema version，owner 信息以及正在执行的 DDL 任务
```sql
ADMIN SHOW DDL;
```

File: session1/chapter7/tidb-ddl-status.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+[tsthght-tia-sql01]admin@sbtest02:04:32>alter table sbtest1 add key idx_c_pad(c, pad);
+^CCtrl-C -- sending "KILL QUERY 15768094" to server ...
+Ctrl-C -- query aborted.
+^CCtrl-C -- exit!
+Aborted
+```
+通过 `ADMIN SHOW DDL` 查询确认，被 `ctrl+c` 之后，该 DDL 依旧正常运行。
+
+```
+[tsthght-tia-sql01]admin@(none)02:04:56>ADMIN SHOW DDL\G
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
tidb> ADMIN SHOW DDL\G
```

File: session1/chapter7/tidb-ddl-status.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+# 取消正在执行中的 DDL 任务
+ADMIN CANCEL DDL JOBS job_id [, job_id] ...;
+# 通过 JOB_ID 恢复表，等价于：RECOVER TABLE table_name
+RECOVER TABLE BY JOB ddl_job_id
+```
+**特别注意**
+在执行一些 DDL 操作时（如 `ADD INDEX` ），由于执行时间较长，不会立即返回执行结果，mysql-client 会处于卡死的状态。此时可以放心的 `ctrl+c` 来终止该连接，不会影响 DDL 的实际执行。DDL 正常耗时可以参考[相关官方文档](https://pingcap.com/docs-cn/stable/faq/tidb/#332-ddl-%E5%9C%A8%E6%AD%A3%E5%B8%B8%E6%83%85%E5%86%B5%E4%B8%8B%E7%9A%84%E8%80%97%E6%97%B6%E6%98%AF%E5%A4%9A%E5%B0%91)
+
+```
+[tsthght-tia-sql01]admin@sbtest02:04:32>alter table sbtest1 add key idx_c_pad(c, pad);
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
tidb> alter table sbtest1 add key idx_c_pad(c, pad);
```

File: session1/chapter7/tidb-ddl-status.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+*************************** 1. row ***************************
+   SCHEMA_VER: 6765
+     OWNER_ID: 828a4567-91a5-4070-b55f-f90702e80e7a
+OWNER_ADDRESS: 10.40.216.9:4000
+ RUNNING_JOBS: ID:8406, Type:add index, State:running, SchemaState:write reorganization, SchemaID:7299, TableID:7380, RowCount:196608, ArgLen:0, start time: 2020-03-07 14:04:53.321 +0800 CST, Err:<nil>, ErrCount:0, SnapshotVersion:415121039339290634
+      SELF_ID: 828a4567-91a5-4070-b55f-f90702e80e7a
+        QUERY: alter table sbtest1 add key idx_c_pad(c, pad)
+1 row in set (0.02 sec)
+```
+除此之外，也可以通过访问 TiDB 提供的 HTTP 接口查看各个 TiDB 节点 `ddl_id`，`lease` 等信息，用法如下：
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
除此之外，也可以通过访问 TiDB 提供的 HTTP 接口查看当前 owner 所在 TiDB，以及各个 TiDB 节点 `ddl_id`、`lease` 等信息，用法如下：
```

File: session1/chapter7/tidb-ddl-status.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+| 属性  | 值  |
+|:------------- |:---------------:|
+| 作用域         | GLOBAL   | 
+| 默认值         | 512   | 
+| 作用   | 控制 DDL 操作失败重试的次数，重试次数超过该值，则取消 DDL 操作   | 
+
+#### 3.2 使用场景
+TiDB 集群中，用户执行的 DDL 操作分两类：普通 DDL 操作和加索引操作。普通 DDL 操作执行时间短，一般秒级就可以执行完成；而加索引操作由于需要回填数据，因此执行时间略长。而在回填数据期间，需要将回填的数据写入 TiKV，对 TiKV 会产生额外的写入压力，从而造成一些性能影响。相关的测试可以参考：[线上负载与 ADD INDEX 相互影响的测试](https://pingcap.com/docs-cn/stable/benchmark/add-index-with-load/#tidb_ddl_reorg_batch_size--32)
+
+TiDB 提供了参数 `tidb_ddl_reorg_worker_cnt` 和 `tidb_ddl_reorg_batch_size` 用来控制回填数据的速度。通过调整参数，可以在业务访问高峰到来时降低 DDL 速度，保证对业务的正常访问无影响；而在业务访问低峰增加DDL速度，从而更快的完成 DDL 任务。
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
TiDB 提供了参数 `tidb_ddl_reorg_worker_cnt` 和 `tidb_ddl_reorg_batch_size` 用来控制回填数据的速度。通过调整参数，可以在业务访问高峰到来时降低 DDL 速度，保证对业务的正常访问无影响；而在业务访问低峰增加 DDL 速度，从而更快的完成 DDL 任务。
```

File: session1/chapter7/tidb-ddl-status.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+参数 `tidb_ddl_reorg_priority` 一旦设置的优先级过高，可能会对正常的 SQL 请求有一定的影响，一般默认值即可。
+
+参数 `tidb_ddl_error_count_limit` 则用来控制重试次数，当发生异常（诸如由于超时、TiKV 无法连接等）时，可进行重试；超过重试次数则终止当前 DDL，一般默认值即可。
+
+### 4  DDL 处理流程
+TiDB-Server 作为SQL的统一入口，DDL 操作也首先经由 TiDB-Server 处理，TiDB 可以多点写入，也就是不同的 TiDB-Server 可以同时接受 DDL 操作请求。
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
TiDB-Server 作为 SQL 的统一入口，DDL 操作也首先经由 TiDB-Server 处理，TiDB 可以多点写入，也就是不同的 TiDB-Server 可以同时接受 DDL 操作请求。
```

File: session1/chapter7/tidb-ddl-status.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+Owner 上的 worker 在处理 `ADD INDEX` 类型 DDL 任务时，涉及到回填数据的过程。该过程会启动 `tidb_ddl_reorg_worker_cnt` 个线程，每次每个线程处理 `tidb_ddl_reorg_batch_size` 大小的数据。因此通过调整这两个参数，可以动态控制 `DDL` 执行速度。
+
+当 `DDL` 任务执行完成之后，会将执行完成的 `DDL` 任务移动至历史任务队列（job history queue）中，方便后续通过命令进行 DDL 任务的历史查询。
+
+完整的 DDL 执行流程如下图所示，更详细的内容，可以参考[DDL 源码解析](https://pingcap.com/blog-cn/tidb-source-code-reading-17/)。
+
+![图片](./images/ddl-workflow.png)
+
+### 5 DDL 变更原理
+TiDB 在线表变更的原理借鉴自论文 《Online, Asynchronous Schema Change in F1》，通过在表结构变更过程中引入额外状态从而实现了一套在线表变更协议，使得集群存在相邻两个版本的schema 时候，不会破坏数据完整性或发生数据异常。因此，在实现上，要求集群中所有schema 版本最多存在两个相邻的版本。
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
TiDB 在线表变更的原理借鉴自论文 《Online, Asynchronous Schema Change in F1》，通过在表结构变更过程中引入额外状态从而实现了一套在线表变更协议，使得集群存在相邻两个版本的 schema 时候，不会破坏数据完整性或发生数据异常。因此，在实现上，要求集群中在同一个表上所有 schema 版本最多存在两个相邻的版本。
```

File: session1/chapter7/tidb-ddl-status.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+| schema版本  | 状态  | 说明  |
+|:------------- |:---------------|:---------------:|
+| schema version 1   | absent   | 添加索引之前。   | 
+| schema version 2   | delete only   | schema 元信息已经修改，但此时对外不可见。内部索引数据不可添加/更新，仅可删除。   | 
+| schema version 3   | write only   | 此时对外不可见，内部索引数据可任意修改。   | 
+| schema version 4   | write reorgnization   | 内部索引数据可任意修改，并且进行索引的回填。   | 
+| schema version 5   | public   | 此时对外可见，索引添加完成。   | 
+
+TiDB 在该论文的基础上又进行了一些优化。例如，在执行 `ADD COLUMN` 类型 DDL 时，TIDB并没有对数据进行回填，而是将最新添加的列的 default 值保存到 schema 的"原始默认值"字段中，在读取阶段如果 TiKV 发现该列值为 `null` 并且"原始默认字段"不为 `null`，则会使用"原始默认字段"对该 `null` 列进行填充，并将填充后的结果从 TiKV 返回。通过这种优化，该 DDL 操作不需要关心表中实际行数，可以更快的完成DDL 变更。
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
TiDB 在该论文的基础上又进行了一些优化。例如，在执行 `ADD COLUMN` 类型 DDL 时，TIDB并没有对数据进行回填，而是将最新添加的列的 default 值保存到 schema 的"原始默认值"字段中，在读取阶段如果 TiKV 发现该列值为 `null` 并且"原始默认字段"不为 `null`，则会使用"原始默认字段"对该 `null` 列进行填充，并将填充后的结果从 TiKV 返回。通过这种优化，该 DDL 操作不需要关心表中实际行数，可以更快的完成 DDL 变更。
```

File: session1/chapter7/tidb-ddl-status.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+
+再例如，一些涉及删除数据的 DDL 操作，诸如：`Drop Index`，`Drop Table`，`Drop Database`，`Truncate Table` 等，在实现上，除了要完成和普通 DDL 变更一样的逻辑外，还需要对待删除的数据进行处理。TiDB 做法是，将这些需要删除的数据记录到 `gc_delete_range` 表中，通过 GC 机制，将对应的数据再进行删除。当然如果是删除某列的 DDL 操作，由于目前是行存储模式，删除列的代价会比较大，所以暂时只是在 schema 上进行删除列的标记，并不会实际删除该列的数据。
+
+更详细的 DDL 实现原理及优化细节可以参考[TiDB DDL architecture](https://github.com/pingcap/tidb/blob/master/docs/design/2018-10-08-online-DDL.md)。
+
+
+
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]: 161-163 这三行可以删除？

File: session1/chapter7/tidb-ddl-status.md (hunk 1 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 2 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed.

File: session1/chapter7/tidb-ddl-status.md (hunk 4 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 5 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+### 2 对 DDL 进行管理
+TiDB 对 MySQL 语法进行了扩展，通过 ADMIN 语句对 DDL 操作进行管理。下面罗列了 DDL 管理的基本命令，各命令返回结果中各字段的详细含义，请参考[官方文档](https://pingcap.com/docs-cn/stable/reference/sql/statements/admin/#admin)
+
+```
+# 查看当前 schema version，owner 信息以及正在执行的 DDL 任务
+ADMIN SHOW DDL;
+# 查看当前 DDL 作业队列中所有结果(包括正在运行的 DDL 任务和等待运行的任务)
+ADMIN SHOW DDL JOBS;
+# 根据 JOB_ID 查询具体的 DDL 语句
+ADMIN SHOW DDL JOB QUERIES job_id [, job_id] ...;
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md:37]: 新的一种更方便的查看 job 方式：`admin show ddl jobs 5 where state!='synced' and db_name='test';` 参考文档：https://pingcap.com/docs-cn/dev/reference/sql/statements/admin/

File: session1/chapter7/tidb-ddl-status.md (hunk 6 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+| schema version 2   | delete only   | schema 元信息已经修改，但此时对外不可见。内部索引数据不可添加/更新，仅可删除。   | 
+| schema version 3   | write only   | 此时对外不可见，内部索引数据可任意修改。   | 
+| schema version 4   | write reorgnization   | 内部索引数据可任意修改，并且进行索引的回填。   | 
+| schema version 5   | public   | 此时对外可见，索引添加完成。   | 
+
+TiDB 在该论文的基础上又进行了一些优化。例如，在执行 `ADD COLUMN` 类型 DDL 时，TIDB并没有对数据进行回填，而是将最新添加的列的 default 值保存到 schema 的"原始默认值"字段中，在读取阶段如果 TiKV 发现该列值为 `null` 并且"原始默认字段"不为 `null`，则会使用"原始默认字段"对该 `null` 列进行填充，并将填充后的结果从 TiKV 返回。通过这种优化，该 DDL 操作不需要关心表中实际行数，可以更快的完成DDL 变更。
+
+再例如，一些涉及删除数据的 DDL 操作，诸如：`Drop Index`，`Drop Table`，`Drop Database`，`Truncate Table` 等，在实现上，除了要完成和普通 DDL 变更一样的逻辑外，还需要对待删除的数据进行处理。TiDB 做法是，将这些需要删除的数据记录到 `gc_delete_range` 表中，通过 GC 机制，将对应的数据再进行删除。当然如果是删除某列的 DDL 操作，由于目前是行存储模式，删除列的代价会比较大，所以暂时只是在 schema 上进行删除列的标记，并不会实际删除该列的数据。
+
+更详细的 DDL 实现原理及优化细节可以参考[TiDB DDL architecture](https://github.com/pingcap/tidb/blob/master/docs/design/2018-10-08-online-DDL.md)。
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
更详细的 DDL 实现原理及优化细节可以参考 [TiDB DDL architecture](https://github.com/pingcap/tidb/blob/master/docs/design/2018-10-08-online-DDL.md)。
```

File: session1/chapter7/tidb-ddl-status.md (hunk 7 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+
+每个 TiDB-Server 上都拥有执行 DDL 任务的 worker 。但是，各个 TiDB-Server 会竞选出唯一一个 Owner 节点来执行实际 DDL 任务，其他竞选失败的 TiDB-Server 节点虽然可以接受 DDL 请求，但是不负责执行 DDL 任务。Owner 会定期对自己的 Owner 身份续租。如果当前 Owner 出现异常，剩余的节点会再次竞选 Owner 。
+
+Owner 节点上的 worker 会从任务队列中依次取出 DDL 任务执行。然而由于 `ADD INDEX` 类型的 DDL 任务在数据量很大情况下执行时间特别长（需要回填数据），从而导致其余的 DDL 操作会被阻塞，TiDB 对该处进行了优化。将任务队列一分为二，一个队列用来存储非 `ADD INDEX` 类型的 DDL 任务，一个队列用来存储 `ADD INDEX` 类型的 DDL 任务。Owner 上的不同的 worker 也会从不同的队列中获取对应的 DDL 任务进行执行。当然，该优化也会引入任务依赖问题。任务依赖问题指的是，同一张表的 DDL 任务，任务编号小的需要先执行。因此在执行 DDL 之前需要对依赖进行检查。
+
+Owner 上的 worker 在处理 `ADD INDEX` 类型 DDL 任务时，涉及到回填数据的过程。该过程会启动 `tidb_ddl_reorg_worker_cnt` 个线程，每次每个线程处理 `tidb_ddl_reorg_batch_size` 大小的数据。因此通过调整这两个参数，可以动态控制 `DDL` 执行速度。
+
+当 `DDL` 任务执行完成之后，会将执行完成的 `DDL` 任务移动至历史任务队列（job history queue）中，方便后续通过命令进行 DDL 任务的历史查询。
+
+完整的 DDL 执行流程如下图所示，更详细的内容，可以参考[DDL 源码解析](https://pingcap.com/blog-cn/tidb-source-code-reading-17/)。
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
完整的 DDL 执行流程如下图所示，更详细的内容，可以参考 [DDL 源码解析](https://pingcap.com/blog-cn/tidb-source-code-reading-17/)。
```

File: session1/chapter7/tidb-ddl-status.md (hunk 8 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 9 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 10 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 11 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 13 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 14 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 3 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md:50]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 12 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md:37]: Thanks , I had added it.

File: session1/chapter7/tidb-ddl-status.md [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+Owner 上的 worker 在处理 `ADD INDEX` 类型 DDL 任务时，涉及到回填数据的过程。该过程会启动 `tidb_ddl_reorg_worker_cnt` 个线程，每次每个线程处理 `tidb_ddl_reorg_batch_size` 大小的数据。因此通过调整这两个参数，可以动态控制 `DDL` 执行速度。
+
+当 `DDL` 任务执行完成之后，会将执行完成的 `DDL` 任务移动至历史任务队列（job history queue）中，方便后续通过命令进行 DDL 任务的历史查询。
+
+完整的 DDL 执行流程如下图所示，更详细的内容，可以参考 [DDL 源码解析](https://pingcap.com/blog-cn/tidb-source-code-reading-17/)。
+
+![图片](./images/ddl-workflow.png)
+
+### 5 DDL 变更原理
+TiDB 在线表变更的原理借鉴自论文 《Online, Asynchronous Schema Change in F1》，通过在表结构变更过程中引入额外状态从而实现了一套在线表变更协议，使得集群存在相邻两个版本的 schema 时候，不会破坏数据完整性或发生数据异常。因此，在实现上，要求集群中所有 schema 版本最多存在两个相邻的版本。
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
TiDB 在线表变更的原理借鉴自论文 《Online, Asynchronous Schema Change in F1》，通过在表结构变更过程中引入额外状态从而实现了一套在线表变更协议，使得集群存在相邻两个版本的 schema 时候，不会破坏数据完整性或发生数据异常。因此，在实现上，要求集群中在同一个表上所有 schema 版本最多存在两个相邻的版本。
```

File: session1/chapter7/tidb-ddl-status.md [hunk 16]
```diff
@@ -0,0 +1,182 @@
+### 1 TiDB DDL 特点
+大多数数据库执行 DDL 操作时，或多或少会对正在访问该数据库的 SQL 产生影响。例如，在执行 DDL 期间可能会有锁表操作，此时访问该表的 SQL 会被阻塞。因此，一般在表结构设计阶段都会尽量避免后续产生 DDL 操作；如果必须执行 DDL操作，也只能选择在业务低峰期操作，尽量减少对线上业务的影响。
+
+TiDB 上的 DDL 操作，不会阻塞任何该数据库上正在执行的 SQL，对业务的 SQL 访问和对 DBA 运维都极为友好，这也是 TiDB 相较于其他数据库产品的一大优势所在。
+
+### 2 对 DDL 进行管理
+TiDB 对 MySQL 语法进行了扩展，通过 ADMIN 语句对 DDL 操作进行管理。下面罗列了 DDL 管理的基本命令，各命令返回结果中各字段的详细含义，请参考[官方 admin 相关文档](https://pingcap.com/docs-cn/stable/reference/sql/statements/admin/#admin)
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
TiDB 对 MySQL 语法进行了扩展，通过 ADMIN 语句对 DDL 操作进行管理。下面罗列了 DDL 管理的基本命令，各命令返回结果中各字段的详细含义，请参考[官方 admin 相关文档](https://pingcap.com/docs-cn/stable/reference/sql/statements/admin/#admin)。
```

File: session1/chapter7/tidb-ddl-status.md [hunk 17]
```diff
@@ -0,0 +1,182 @@
...
+ADMIN SHOW DDL;
+```
+
+- 查看当前未执行完成的 DDL 任务(包括正在运行的 DDL 任务和等待运行的任务)以及最近 NUM 条(默认 10 )已经执行完成的 DDL 任务
+
+```
+ADMIN SHOW DDL JOBS [NUM] [WHERE where_condition];
+
+# 例如，显示当前未完成的 DDL 任务，以及最近 5 条已经执行完成的 DDL 任务
+ADMIN SHOW DDL JOBS 5;
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md:25]: 是不是这里也添加一行，跟下面的操作隔开？跟 line18 格式一致一些

File: session1/chapter7/tidb-ddl-status.md [hunk 18]
```diff
@@ -0,0 +1,182 @@
...
+ADMIN CANCEL DDL JOBS job_id [, job_id] ...;
+```
+
+- 通过 JOB_ID 恢复表，等价于：RECOVER TABLE table_name
+
+```
+RECOVER TABLE BY JOB ddl_job_id;
+```
+**特别注意**
+在执行一些 DDL 操作时（如 `ADD INDEX` ），由于执行时间较长，不会立即返回执行结果，mysql-client 会处于卡死的状态。此时可以放心的 `ctrl+c` 来终止该连接，不会影响 DDL 的实际执行。DDL 正常耗时可以参考[相关官方文档](https://pingcap.com/docs-cn/stable/faq/tidb/#332-ddl-%E5%9C%A8%E6%AD%A3%E5%B8%B8%E6%83%85%E5%86%B5%E4%B8%8B%E7%9A%84%E8%80%97%E6%97%B6%E6%98%AF%E5%A4%9A%E5%B0%91)
```
Comment (zimulala) [inline session1/chapter7/tidb-ddl-status.md]:
```suggestion
在执行一些 DDL 操作时（如 `ADD INDEX` ），由于执行时间较长，不会立即返回执行结果，mysql-client 会处于卡死的状态。此时可以放心的 `ctrl+c` 来终止该连接，不会影响 DDL 的实际执行。DDL 正常耗时可以参考[相关官方文档](https://pingcap.com/docs-cn/stable/faq/tidb/#332-ddl-%E5%9C%A8%E6%AD%A3%E5%B8%B8%E6%83%85%E5%86%B5%E4%B8%8B%E7%9A%84%E8%80%97%E6%97%B6%E6%98%AF%E5%A4%9A%E5%B0%91)。
```

File: session1/chapter7/tidb-ddl-status.md (hunk 15 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 16 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 18 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md]: Thanks, I had fixed it.

File: session1/chapter7/tidb-ddl-status.md (hunk 17 above)
Comment (tsthght) [reply session1/chapter7/tidb-ddl-status.md:25]: Thanks, I had fixed it.

**Review by zimulala — COMMENTED** [review]
LGTM

**Review by winkyao — APPROVED** [review]
LGTM

**Pull request merged**